src/apps/filters/h3.c

```diff
@@ -727,6 +727,468 @@ SUBCOMMAND(localIjToCell,
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, 1-15 inclusive, excluding 0 as it can "
+                      "never be a child"};
+    Arg *args[] = {&cellToCenterChildArg, &helpArg, &cellArg, &resArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    H3Index centerChild;
+    int valid = H3_EXPORT(isValidCell)(cell);
+    if (valid == 0) {
+        return E_CELL_INVALID;
+    }
+    H3Error err = H3_EXPORT(cellToCenterChild)(cell, res, &centerChild);
+    if (err) {
+        return err;
+    }
+    h3Println(centerChild);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(
+    cellToChildPos,
+    "Returns the position of the child cell within an ordered list of all "
+    "children of the cell's parent at the specified child resolution") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    int res = 0;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, 1-15 inclusive, excluding 0 as it can "
+                      "never be a child"};
+    Arg *args[] = {&cellToChildPosArg, &helpArg, &cellArg, &resArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    int64_t len = 0;
+    H3Error err = H3_EXPORT(cellToChildPos)(cell, res, &len);
+    if (err) {
+        return err;
+    }
+    printf("%" PRId64, len);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(childPosToCell,
+           "Returns the child cell at a given position and resolution within "
+           "an ordered list of all children of the parent cell") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    int res = 0;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText =
+                      "Resolution, 1-15 inclusive, excluding 0 as it can "
+                      "never be a child"};
+    int64_t pos;
+    Arg posArg = {
+        .names = {"-p", "--position"},
+        .required = true,
+        .scanFormat = "%" PRIi64,
+        .valueName = "pos",
+        .value = &pos,
+        .helpText =
+            "The child position within the set of children of the parent cell"};
+    Arg *args[] = {&childPosToCellArg, &helpArg, &cellArg, &resArg, &posArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    H3Index child;
+    int valid = H3_EXPORT(isValidCell)(cell);
+    if (valid == 0) {
+        return E_CELL_INVALID;
+    }
+    H3Error err = H3_EXPORT(childPosToCell)(pos, cell, res, &child);
+    if (err) {
+        return err;
+    }
+    h3Println(child);
+    return E_SUCCESS;
+}
+
+H3Index *readCellsFromFile(FILE *fp, char *buffer, int *totalCells) {
```

Comment:
```suggestion
H3Index *readCellsFromFile(FILE *fp, char *buffer, size_t *totalCells) {
```
otherwise may run into an issue after a few gigabytes